flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSpillingInfoProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid;
+
+import java.util.Deque;
+import java.util.List;
+
+/** This component is responsible for providing some information needed for the spill decision. */
+public interface HsSpillingInfoProvider {
+    /**
+     * Get the number of downstream consumers.
+     *
+     * @return Number of subpartitions.
+     */
+    int getNumSubpartitions();
+
+    /**
+     * Get all downstream consumption progress.
+     *
+     * @return A list containing all downstream consumption progress, if the downstream subpartition
+     *     view has not been registered, the corresponding return value is -1.
+     */
+    List<Integer> getConsumptionProgress();
```

Comment:
"Consuming progress" is ambiguous. It can be "last consumed buffer index", or "next buffer index to consume".

Reply:
renamed to `getNextBufferIndexToConsume`